packages/components/src/utils/__tests__/getWordsFromPermalink.test.ts

```diff
@@ -0,0 +1,66 @@
+import { describe, expect, it } from "vitest"
+
+import { getWordsFromPermalink } from "~/utils"
+
+describe("getWordsFromPermalink", () => {
+  it("should trim out all symbols and return the permalink as a space separated sentence for single level permalinks", () => {
+    // Arrange
+    const singleLevelPermalink = "/this-._single=level|"
+    const expected = "this+single+level"
+
+    // Act
+    const actual = getWordsFromPermalink(singleLevelPermalink)
+
+    // Assert
+    expect(actual).toBe(expected)
+  })
+
+  it("should trim out all symbols and return the last section as a space separated sentence for nested permalinks", () => {
+    // Arrange
+    const nestedPermalink = "/nested/deeply/this-._nest'fff=level|"
+    const expected = "this+nest+fff+level"
+
+    // Act
+    const actual = getWordsFromPermalink(nestedPermalink)
+
+    // Assert
+    expect(actual).toBe(expected)
+  })
+
+  it("should preserve `=` in the original permalink", () => {
+    // Arrange
+    const singleLevelPermalink = "/this-._single=level|"
+    const expected = "this+single+level"
+
+    // Act
+    const actual = getWordsFromPermalink(singleLevelPermalink)
+
+    // Assert
+    expect(actual).toBe(expected)
+  })
```

Comment:
unsure if i understand the test description correctly but `=` isn't preserve but converted?

Reply:
this is my bad, i was flipflopping on whether we should keep this due to query params but i decided teh `=` wasn't going to impact the search results much - i'll just remove this test and add a `=` character ot the other permlainks